--- php-4.4.7/ext/gd/libgd/gdft.c	2007-03-10 13:51:07.000000000 +0100
+++ php-4.4.7.org/ext/gd/libgd/gdft.c	2007-05-30 21:08:19.917066032 +0200
@@ -750,10 +750,8 @@
 		  /* find antialised color */
 	
 		  tc_key.bgcolor = *pixel;
-		  gdMutexLock(gdFontCacheMutex);
 		  tc_elem = (tweencolor_t *) gdCacheGet (tc_cache, &tc_key);
 		  *pixel = tc_elem->tweencolor;
-		  gdMutexUnlock(gdFontCacheMutex);
 		}
 	    }
 	}
@@ -771,30 +769,39 @@
 
 void gdFontCacheShutdown()
 {
+	gdMutexLock(gdFontCacheMutex);
+
 	if (fontCache) {
-		gdMutexLock(gdFontCacheMutex);
 		gdCacheDelete(fontCache);
 		fontCache = NULL;
-		gdMutexUnlock(gdFontCacheMutex);
-		gdMutexShutdown(gdFontCacheMutex);
 		FT_Done_FreeType(library);
 	}
+
+	gdMutexUnlock(gdFontCacheMutex);
 }
 
 void gdFreeFontCache()
 {
 	gdFontCacheShutdown();
 }
-  
+
+void gdFontCacheMutexSetup()
+{
+        gdMutexSetup(gdFontCacheMutex);
+}
+
+void gdFontCacheMutexShutdown()
+{
+        gdMutexShutdown(gdFontCacheMutex);
+}
+
 int gdFontCacheSetup(void)
 {
 	if (fontCache) {
 		/* Already set up */
 		return 0;
 	}
-	gdMutexSetup(gdFontCacheMutex);
 	if (FT_Init_FreeType(&library)) {
-		gdMutexShutdown(gdFontCacheMutex);
 		return -1;
 	}
 	fontCache = gdCacheCreate (FONTCACHESIZE, fontTest, fontFetch, fontRelease);
@@ -856,15 +863,16 @@
 
 	/***** initialize font library and font cache on first call ******/
 
+	gdMutexLock(gdFontCacheMutex);
 	if (!fontCache) {
 		if (gdFontCacheSetup() != 0) {
 			gdCacheDelete(tc_cache);
+			gdMutexUnlock(gdFontCacheMutex);
 			return "Failure to initialize font library";
 		}
 	}
 	/*****/
 	
-	gdMutexLock(gdFontCacheMutex);
 	/* get the font (via font cache) */
 	fontkey.fontlist = fontlist;
 	fontkey.library = &library;
