--- php-4.3.3RC3/ext/qtdom/config.m4~	2002-11-18 06:06:37.000000000 +0100
+++ php-4.3.3RC3/ext/qtdom/config.m4	2003-08-17 11:40:07.000000000 +0200
@@ -11,6 +11,9 @@
       if test -f $QTDIR/include/qdom.h; then
         QTDOM_LIBDIR=$QTDIR/lib
         QTDOM_INCDIR=$QTDIR/include
+      elif test -f /usr/include/qt/qdom.h; then
+        QTDOM_LIBDIR=/usr/lib
+        QTDOM_INCDIR=/usr/include/qt
       elif test -f /usr/lib/qt2/include/qdom.h; then
         QTDOM_LIBDIR=/usr/lib
         QTDOM_INCDIR=/usr/lib/qt2/include
@@ -28,7 +31,7 @@
     AC_MSG_ERROR([qdom.h not found.])
   fi
 
-  PHP_ADD_LIBRARY_WITH_PATH(qt, $QTDOM_LIBDIR, QTDOM_SHARED_LIBADD)
+  PHP_ADD_LIBRARY_WITH_PATH(qt-mt, $QTDOM_LIBDIR, QTDOM_SHARED_LIBADD)
   PHP_ADD_INCLUDE($QTDOM_INCDIR)
   PHP_NEW_EXTENSION(qtdom, qtdom.c qtdom_qt.cpp, $ext_shared)
   PHP_SUBST(QTDOM_SHARED_LIBADD)
--- php-4.4.9/ext/qtdom/qtdom_qt.h.orig	2001-01-04 13:29:07.000000000 +0100
+++ php-4.4.9/ext/qtdom/qtdom_qt.h	2020-10-22 21:19:25.667779334 +0200
@@ -62,6 +62,6 @@
 struct qdom_doc *qdom_do_init( const char *arg );
 void qdom_do_free( struct qdom_doc *doc );
 
-struct qdom_message *g_qdom_message_log;
+extern struct qdom_message *g_qdom_message_log;
 
 #endif // QDOM_QT_H
--- php-4.4.9/ext/qtdom/qtdom_qt.cpp.orig	2001-01-04 13:30:19.000000000 +0100
+++ php-4.4.9/ext/qtdom/qtdom_qt.cpp	2020-10-22 21:20:56.769184131 +0200
@@ -22,6 +22,7 @@
 #include <qglobal.h>
 
 struct qdom_node *qdom_to_node( QDomNode *node );
+struct qdom_message *g_qdom_message_log;
 
 /*!
   Initialises certain global variables, they are:
