--- php-4.3.9RC2.orig/ext/mnogosearch/config.m4	2004-09-01 18:13:53.495238432 +0200
+++ php-4.3.9RC2/ext/mnogosearch/config.m4	2004-09-01 18:16:52.696995624 +0200
@@ -46,8 +46,10 @@
     PHP_ADD_INCLUDE($MNOGOSEARCH_INCDIR)
     
     AC_DEFINE(HAVE_MNOGOSEARCH,1,[ ])
+    MNOGOSEARCH_CFLAGS="-DNOT_USED_OR_REPLACED"
     
-    PHP_NEW_EXTENSION(mnogosearch, php_mnogo.c, $ext_shared)
+    PHP_SUBST(MNOGOSEARCH_CFLAGS)
+    PHP_NEW_EXTENSION(mnogosearch, php_mnogo.c, $ext_shared,, $MNOGOSEARCH_CFLAGS)
     PHP_SUBST(MNOGOSEARCH_SHARED_LIBADD)
     
   fi
--- php-4.4.9/ext/mnogosearch/php_mnogo.c~	2009-06-25 22:11:26.000000000 +0300
+++ php-4.4.9/ext/mnogosearch/php_mnogo.c	2009-06-25 22:14:22.544491038 +0300
@@ -901,10 +901,14 @@
 			if (UdmSynonymListLoad(Agent->Conf,val)) {
 				php_error_docref(NULL TSRMLS_CC, E_WARNING, "%s",Agent->Conf->errstr);
 				RETURN_FALSE;
+#if UDM_VERSION_ID >= 30300
+			} else UdmSynonymListSort(&(Agent->Conf->Synonym));
+#else
 #if UDM_VERSION_ID >= 30204
 			} else UdmSynonymListSort(&(Agent->Conf->Synonyms));
 #else
 			} else UdmSynonymListSort(&(Agent->Conf->SynList));
+#endif
 #endif			
 			break;
 			
@@ -1819,7 +1823,11 @@
 		if (UdmSynonymListLoad(Agent->Conf,val)) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "%s",Agent->Conf->errstr);
 			RETURN_FALSE;
+#if UDM_VERSION_ID >= 30300
+		} else UdmSynonymListSort(&(Agent->Conf->Synonym));
+#else
 		} else UdmSynonymListSort(&(Agent->Conf->Synonyms));
+#endif
 	} else if (!strcasecmp(var,"Stopwordfile")) {
 		if (UdmStopListLoad(Agent->Conf,val)) {
 			php_error_docref(NULL TSRMLS_CC, E_WARNING, "%s", Agent->Conf->errstr);
