--- php-4.2.2/ext/mysql/config.m4.orig	Fri Aug  9 23:49:00 2002
+++ php-4.2.2/ext/mysql/config.m4	Sat Aug 10 00:05:40 2002
@@ -49,6 +49,7 @@
 [  --with-mysql[=DIR]      Include MySQL support. DIR is the MySQL base directory.
                           If unspecified, the bundled MySQL library will be used.], yes)
 
+PHP_REQUIRE_CXX
 if test "$PHP_MYSQL" != "no"; then
   AC_DEFINE(HAVE_MYSQL, 1, [Whether you have MySQL])
   if test "$PHP_SAFE_MODE" = "yes"; then
--- php-4.4.0/configure.in~	2005-07-25 00:22:36.000000000 +0300
+++ php-4.4.0/configure.in	2005-07-25 01:05:09.000000000 +0300
@@ -844,6 +844,9 @@
 dnl -------------------------------------------------------------------------
 
 if test "$PHP_VERSIONING" = "yes"; then
+  if test -n "$PHP_MODULES"; then
+    AC_MSG_ERROR([--enable-versioning cannot be used with shared modules])
+  fi
   test -z "$PHP_SYM_FILE" && PHP_SYM_FILE="$abs_srcdir/sapi/$PHP_SAPI/php.sym"
   if test -f "$PHP_SYM_FILE"; then
     EXTRA_LDFLAGS="-export-symbols $PHP_SYM_FILE $EXTRA_LDFLAGS"
