--- php-4.3.0/configure.in.orig	Fri Dec 27 05:07:18 2002
+++ php-4.3.0/configure.in	Thu Jan  2 20:34:43 2003
@@ -288,7 +288,6 @@
 PHP_CHECK_FUNC(htonl, socket)
 PHP_CHECK_FUNC(gethostname, nsl)
 PHP_CHECK_FUNC(gethostbyaddr, nsl)
-PHP_CHECK_FUNC(yp_get_default_domain, nsl)
 
 PHP_CHECK_FUNC(dlopen, dl)
 if test "$ac_cv_func_dlopen" = "yes"; then
--- php.org/ext/ldap/config.m4.org	2006-08-10 14:23:59.583381250 +0000
+++ php/ext/ldap/config.m4	2006-08-10 14:24:40.013908000 +0000
@@ -113,8 +113,9 @@
   dnl Save original values
   _SAVE_CPPFLAGS=$CPPFLAGS
   _SAVE_LDFLAGS=$LDFLAGS
-  LDFLAGS="$LDFLAGS $LDAP_SHARED_LIBADD"
+  LDFLAGS="$LDFLAGS"
   CPPFLAGS="$CPPFLAGS -I$LDAP_INCDIR"
+  LIBS="$LIBS $LDAP_SHARED_LIBADD"
 
   dnl Check for 3 arg ldap_set_rebind_proc
   AC_CACHE_CHECK([for 3 arg ldap_set_rebind_proc], ac_cv_3arg_setrebindproc,
@@ -138,4 +139,5 @@
   dnl Restore original values
   CPPFLAGS=$_SAVE_CPPFLAGS
   LDFLAGS=$_SAVE_LDFLAGS
+  LIBS=$_SAVE_LIBS
 fi 
