diff -ur php-4.4.4.orig/ext/imap/config.m4 php-4.4.4/ext/imap/config.m4
--- php-4.4.4.orig/ext/imap/config.m4	2006-12-31 20:11:37.000000000 +0100
+++ php-4.4.4/ext/imap/config.m4	2006-12-31 20:12:01.000000000 +0100
@@ -126,6 +126,11 @@
     AC_EGREP_HEADER(mail_fetch_overview_sequence, $IMAP_INC_DIR/mail.h, [
       AC_DEFINE(HAVE_IMAP2004,1,[ ])
     ])
+    
+    dnl Check for new version of the utf8_mime2text() function
+    AC_EGREP_HEADER(mail_append_set, $IMAP_INC_DIR/mail.h, [
+      AC_DEFINE(HAVE_NEW_MIME2TEXT,1,[ ])
+    ])
 
     dnl Check for c-client version 2001
     old_CPPFLAGS=$CPPFLAGS
diff -ur php-4.4.4.orig/ext/imap/php_imap.c php-4.4.4/ext/imap/php_imap.c
--- php-4.4.4.orig/ext/imap/php_imap.c	2006-12-31 20:11:37.000000000 +0100
+++ php-4.4.4/ext/imap/php_imap.c	2006-12-31 20:13:45.000000000 +0100
@@ -71,7 +71,11 @@
 void rfc822_date(char *date);
 char *cpystr(const char *str);
 char *cpytxt(SIZEDTEXT *dst, char *text, unsigned long size);
+#ifndef HAVE_NEW_MIME2TEXT
 long utf8_mime2text(SIZEDTEXT *src, SIZEDTEXT *dst);
+#else
+long utf8_mime2text (SIZEDTEXT *src, SIZEDTEXT *dst, long flags);
+#endif
 unsigned long find_rightmost_bit(unsigned long *valptr);
 void fs_give(void **block);
 void *fs_get(size_t size);
@@ -2061,7 +2065,11 @@
 	dest.size = 0;
 
 	cpytxt(&src, Z_STRVAL_PP(str), Z_STRLEN_PP(str));
+#ifndef HAVE_NEW_MIME2TEXT
 	utf8_mime2text(&src, &dest);
+#else
+	utf8_mime2text(&src, &dest, U8T_CANONICAL);
+#endif
 	RETURN_STRINGL(dest.data, strlen(dest.data), 1);
 }
 /* }}} */
